modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2MessageContext.java

```diff
@@ -511,17 +512,22 @@ public Log getServiceLog() {
         if (serviceLog != null) {
             return serviceLog;
         } else {
+            String serviceLoggerName;
             String serviceName = (String) getProperty(SynapseConstants.PROXY_SERVICE);
+            // Added REST API service logger resolver to fix ESBJAVA-5216.
+            if (serviceName == null) {
+                serviceName = (String) getProperty(RESTConstants.SYNAPSE_REST_API);
+            }
             if (serviceName != null && synCfg.getProxyService(serviceName) != null) {
```

Comment:
I think we can arrange these checks in a way that we don't have to check whether service is a proxy or an api two times. For an example we can nest synCfg.getProxyService(serviceName) with serviceName != null